File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-1.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        first = 0
        second = 1
        while second < len(nums):
            if nums[first] == nums[second]:
                second+=1
            else:
                first+=1
                nums[first] = nums[second]
                second+=1
        return first+1        

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int,int> freq;
        priority_queue<pair<int,int>, vector<pair<int,int>>,
        greater<pair<int,int>>> minheap;
        vector<int> result;

        for(int x:nums)
        {
            freq[x]++;
        }
        for(auto &x: freq)
        {
            minheap.push({x.second, x.first});
            if(minheap.size()>k)
            {
                minheap.pop();
            }
        }
        while(!minheap.empty())
        {
            result.push_back(minheap.top().second);
            minheap.pop();

        }





       return result; 
    }
};
